rmw_implementation/test/benchmark/benchmark_symbols.cpp

```diff
@@ -23,8 +23,8 @@ using performance_test_fixture::PerformanceTest;
 BENCHMARK_F(PerformanceTest, prefetch_symbols)(benchmark::State & st)
 {
   for (auto _ : st) {
-    std::shared_ptr<rcpputils::SharedLibrary> lib = load_library();
     prefetch_symbols();
+    unload_library();
```

Comment:
@cottsay meta: are we benchmarking the OS dynamic linker?

Reply:
I think it's less about benchmarking the linker, and more about benchmarking how long it takes to load an RMW. The tests are currently show inconsistencies in this process that are hinting at some nondeterminism somewhere, and I'm also seeing that Cyclone DDS is taking 10 times as long as Fast-DDS to perform this prefetch operation - something we'll absolutely want to investigate.

It's less about the micro-scale testing of "how good is the OS dynamic linker performing", and more about determining how much this costs in total, and comparing between RMWs.

Reply:
> I'm also seeing that Cyclone DDS is taking 10 times as long as Fast-DDS to perform this prefetch operation - something we'll absolutely want to investigate.

Uh, really? That's both unexpected _and_ interesting. Could it be static storage initialization within `rmw_cyclonedds_cpp` or any of its dependencies?

Reply:
On my local machine, yeah. I mean, we're talking about 1/100 of a second or so, but it's still interesting. Connext is pretty bad, too. The numbers look better on the CI machine though.

Let's wait for the tests running regularly in the dedicated benchmark job before we launch any investigations. I just wanted to call it out the emphasize why I think this is a really great benchmark to track.